test/unit/cmock_header_parser_test.rb

```diff
@@ -1997,4 +2029,31 @@
     assert_equal(0, @parser.count_number_of_pairs_of_braces_in_function(bad_source_2))
   end
 
+  it "Transform inline functions takes user provided patterns into account" do
+    source =
+      "static inline int staticinlinefunc(struct my_struct *s)\n" + # 'normal' inline pattern
+      "{\n" +
+      "    return s->a;\n" +
+      "}\n" +
+      "static __inline__ int dummy_func_2(int a, char b, float c) {\n" + # First user pattern
+      "	c += 3.14;\n" +
+      "	b -= 32;\n" +
+      "	return a + (int)(b) + (int)c;\n" +
+      "}\n" +
+      "static __inline__ __attribute__ ((always_inline)) uint16_t attributealwaysinlinefuncname(void) {\n" + # Second user pattern
```

Comment:
Maybe order here is important for this to fail. Can you put this before the `static __inline__` only, so that we see if it fails in that combination?

Reply:
I tried to move the `attributealwasyinlinefuncname` before `dummy_func_2` but the test still passes.
As stated in the MR, if I move `static __inline__` up in the array declared on line 2055:
```
@parser.inline_function_patterns = ['static __inline__', 'static __inline__ __attribute__ \(\(always_inline\)\)', 'static inline']
```
Then the test fails because `__attribute__((always_inline))` is not removed but it's still another error than the one you are seeing I think.

Reply:
Ok, if test still passes then there is something more with my issue. Anyway - it is not blocking merge of this.